synapse/replication/tcp/resource.py

```diff
@@ -199,33 +199,28 @@ async def _run_notifier_loop(self) -> None:
                             # The token has advanced but there is no data to
                             # send, so we send a `POSITION` to inform other
                             # workers of the updated position.
-                            if stream.NAME == EventsStream.NAME:
-                                # XXX: We only do this for the EventStream as it
-                                # turns out that e.g. account data streams share
-                                # their "current token" with each other, meaning
-                                # that it is *not* safe to send a POSITION.
```

Comment:
Does this changeset workaround/prevent the problem described here?

Reply:
Argh, I'm awful at linking together PRs. We fixed this in https://github.com/matrix-org/synapse/pull/14826 and https://github.com/matrix-org/synapse/pull/14833